src/data/geo/linestring.js

```diff
@@ -0,0 +1,79 @@
+import Geometry from './geometry'
+import SpatialReferenceSystem from './spatial-reference-system'
+import { GeoTypes } from './constants'
+
+const GEOJSON_TYPE = GeoTypes.LINE_STRING
+const WKT_TYPE = GEOJSON_TYPE.toUpperCase()
+
+const DIFFERENT_SRS_ERROR_MSG = (
+  'Some of the points were created in the Spatial Reference System (SRS) that differs from the passed argument \'srs\'.'
+)
+
+const POINTS_NUMBER_ERROR_MSG = 'The number of points must be greater than two.'
+
+const NOT_ARRAY_ERROR_MSG = 'Points argument should be an array'
+
+const validate = (points, srs) => {
+  if (!Array.isArray(points)) {
+    throw new Error(NOT_ARRAY_ERROR_MSG)
+  }
+
+  if (points.length < 2) {
+    throw new Error(POINTS_NUMBER_ERROR_MSG)
+  }
+
+  for (const point of points) {
+    if (point.getSRS() !== srs) {
+      throw new Error(DIFFERENT_SRS_ERROR_MSG)
+    }
+  }
+}
+
+class LineString extends Geometry {
+  constructor(points, srs) {
+    srs = srs || SpatialReferenceSystem.DEFAULT
```

Comment:
No, because we must verify first that if the user explicitly pass srs and this srs is empty or undefined, then this action must be prevented by throwing an error. If only srs just was not passed, then the default value is used.

Reply:
not sure I understand what you mean. 

I can see that every `Line|Point|Polygon` class has `srs = srs || SpatialReferenceSystem.DEFAULT` right before calling `super constructor` of Geometry class, so it can be relocated inside `Geometry`, can't it?

or you mean that `Geometry` have to validate `srs`? if so, why do we need that?

Reply:
Was moved to Geometry class